lib/utils.js

```diff
@@ -219,8 +218,13 @@ var Utils = module.exports = {
   },
 
   toDefaultValue: function(value) {
-    if (lodash.isFunction(value)) {
-      return value();
+    if (typeof value === 'function') {
+      var tmp = value();
+      if (tmp instanceof DataTypes.ABSTRACT) {
```

Comment:
Hm, right now there's no call that could reach that point and I doubt that it's possible to know when to call new in toDefaultValue.